packages/react-components/react-virtualizer/src/components/VirtualizerScrollViewDynamic/VirtualizerScrollViewDynamic.types.ts

```diff
@@ -32,6 +34,14 @@ export type VirtualizerScrollViewDynamicProps = ComponentProps<Partial<Virtualiz
      * Will act as a row or column indexer depending on Virtualizer settings.
      */
     children: VirtualizerChildRenderFunction;
+    /**
+     * Imperative ref contains our scrollTo index functionality for user control.
+     */
+    imperativeScrollRef?: RefObject<ScrollToInterface>;
+    /**
+     * Callback to notify user once index has rendered.
+     */
+    didScrollToIndex?: (index: number) => void;
```

Comment:
since the scroll operations should only happen when the user triggers this once, why not make `scrollTo` take as argument this callback?

```tsx
scrollTo(5, 'auto', () => { /* do logic here */})
```

The operation users want after an imperative scroll might change depending on the use case, and I don't think changing prop is a good idea to do this.

Reply:
Good call, I like this a lot.

I think we will still only enable a single index to be 'tracked', so if a scrollTo call is called before this one completes it will clobber the previous, if that makes sense.